src/main/java/com/faforever/client/logging/LoggingService.java

```diff
@@ -69,10 +71,17 @@ public void afterPropertiesSet() throws IOException, InterruptedException, Joran
     JavaFxUtil.addAndTriggerListener(developerPrefs.logLevelProperty(), new WeakChangeListener<>(logLevelChangeListener));
   }
 
-  public Path getNewGameLogFile(int gameUID) {
+  public Path getNewGameLogFile(int gameUID) {return getNewGameLogFile(gameUID, false);}
+
+  public Path getNewGameLogFile(int gameUID, boolean replay) {
+    if (replay) {
+      LOG_PATTERN = REPLAY_LOG_PATTERN;
+    } else {
+      LOG_PATTERN = GAME_LOG_PATTERN;
+    }
     try (Stream<Path> listOfLogFiles = Files.list(operatingSystem.getLoggingDirectory())) {
       listOfLogFiles
-          .filter(logPath -> GAME_LOG_PATTERN.matcher(logPath.getFileName().toString()).matches())
+          .filter(logPath -> LOG_PATTERN.matcher(logPath.getFileName().toString()).matches())
```

Comment:
We shouldn't be assigning a static variable here as this makes it not thread safe when people can run a replay and a game at the same time.

We should instead have two public methods with one private method that takes the pattern as a parameter